Traceback (most recent call last):
  File "/Users/jfoley19/miniconda3/envs/jbook/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/jfoley19/miniconda3/envs/jbook/lib/python3.11/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jfoley19/miniconda3/envs/jbook/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jfoley19/miniconda3/envs/jbook/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/jfoley19/miniconda3/envs/jbook/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/Users/jfoley19/miniconda3/envs/jbook/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/jfoley19/miniconda3/envs/jbook/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import numpy as np
from scipy import constants


# define zeroth-order ground state energy in Joules
E_1_zo = -2.179e-18

# define zeroth-order first excited state(s) energy in Joules
E_2_zo = -5.447e-19

# get hbar from scipy constants
hbar = constants.hbar

# get electron charge from scipy constants
e = constants.e

# get electron mass from scipy constants
me = constants.m_e

# value of the magnetic field in Tesla (SI units)
B = 200

# get first order correction for n = 2, l = 1, m = 1
E_211_fo = -e * hbar * B / (2 * me)

# get first order correction for n = 2, l = 1, m = 0
E_210_fo = 0

# get first order correction for n = 2, l = 1, m = -1
E_21m1_fo = e * hbar * B / (2 * me)

# print results
E_211 = E_2_zo + E_211_fo
E_210 = E_2_zo + E_210_fo
E_21m1 = E_2_zo + E_21m1_fo

print("Printing Energies to first order")
print(f"E_2,1,1  = {E_211} J")
print(f"E_2,1,0  = {E_210} J")
print(f"E_2,1,-1 = {E_21m1} J")


------------------


[31m---------------------------------------------------------------------------[39m
[31mModuleNotFoundError[39m                       Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[1][39m[32m, line 2[39m
[32m      1[39m [38;5;28;01mimport[39;00m[38;5;250m [39m[34;01mnumpy[39;00m[38;5;250m [39m[38;5;28;01mas[39;00m[38;5;250m [39m[34;01mnp[39;00m
[32m----> [39m[32m2[39m [38;5;28;01mfrom[39;00m[38;5;250m [39m[34;01mscipy[39;00m[38;5;250m [39m[38;5;28;01mimport[39;00m constants
[32m      5[39m [38;5;66;03m# define zeroth-order ground state energy in Joules[39;00m
[32m      6[39m E_1_zo = -[32m2.179e-18[39m

[31mModuleNotFoundError[39m: No module named 'scipy'

